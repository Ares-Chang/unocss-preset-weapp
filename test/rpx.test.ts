import { createGenerator } from '@unocss/core'
import { describe, expect, test } from 'vitest'
import presetMini from '../src'

const taroH5_750 = createGenerator({
  presets: [
    presetMini({
      isH5: true,
      platform: 'taro',
    }),
  ],
})

const taroWe_750_W4 = createGenerator({
  presets: [
    presetMini({
      isH5: false,
      platform: 'taro',
    }),
  ],
})

const taroH5_375_W4 = createGenerator({
  presets: [
    presetMini({
      isH5: true,
      platform: 'taro',
      designWidth: 375,
      deviceRatio: {
        375: 2 / 1,
      },
    }),
  ],
})

const taroWe_750_W5 = createGenerator({
  presets: [
    presetMini({
      isH5: false,
      platform: 'taro',
      taroWebpack: 'webpack5',
    }),
  ],
})

const taroH5_375_W5 = createGenerator({
  presets: [
    presetMini({
      isH5: true,
      platform: 'taro',
      designWidth: 375,
      deviceRatio: {
        375: 2 / 1,
      },
      taroWebpack: 'webpack5',
    }),
  ],
})

const taroWe_375 = createGenerator({
  presets: [
    presetMini({
      isH5: false,
      platform: 'taro',
      designWidth: 375,
      deviceRatio: {
        375: 2 / 1,
      },
    }),
  ],
})

const uniapp_750 = createGenerator({
  presets: [
    presetMini({
      isH5: true,
      platform: 'uniapp',
    }),
  ],
})

const size = [
  // 'w-100px',
  // 'w-100',
  // 'w-100rpx',

  'text-base',
  'text-16px',
  // 'm-6',
]

describe('rpx-tranform', () => {
  test('taroH5_750', async () => {
    const code = size.join(' ')
    const { css } = await taroH5_750.generate(code)
    expect(css).toMatchSnapshot()
  })

  test('taroWe_750_W4', async () => {
    const code = size.join(' ')
    const { css } = await taroWe_750_W4.generate(code)
    expect(css).toMatchSnapshot()
  })

  test('taroH5_375_W4', async () => {
    const code = size.join(' ')
    const { css } = await taroH5_375_W4.generate(code)
    expect(css).toMatchSnapshot()
  })

  test('taroWe_750_W5', async () => {
    const code = size.join(' ')
    const { css } = await taroWe_750_W5.generate(code)
    expect(css).toMatchSnapshot()
  })

  test('taroH5_375_W5', async () => {
    const code = size.join(' ')
    const { css } = await taroH5_375_W5.generate(code)
    expect(css).toMatchSnapshot()
  })

  test('taroWe_375', async () => {
    const code = size.join(' ')
    const { css } = await taroWe_375.generate(code)
    expect(css).toMatchSnapshot()
  })

  test('uniapp_750', async () => {
    const code = size.join(' ')
    const { css } = await uniapp_750.generate(code)
    expect(css).toMatchSnapshot()
  })
})
