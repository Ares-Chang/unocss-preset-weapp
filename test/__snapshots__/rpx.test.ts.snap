// Vitest Snapshot v1

exports[`rpx-tranform > taroH5_375 1`] = `
"/* layer: preflights */
*,::before,::after{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgba(0,0,0,0);--un-ring-shadow:0 0 rgba(0,0,0,0);--un-shadow-inset:var(--un-empty,/*!*/ /*!*/);--un-shadow:0 0 rgba(0,0,0,0);--un-ring-inset:var(--un-empty,/*!*/ /*!*/);--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgba(147,197,253,0.5);}::backdrop{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgba(0,0,0,0);--un-ring-shadow:0 0 rgba(0,0,0,0);--un-shadow-inset:var(--un-empty,/*!*/ /*!*/);--un-shadow:0 0 rgba(0,0,0,0);--un-ring-inset:var(--un-empty,/*!*/ /*!*/);--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgba(147,197,253,0.5);}
/* layer: default */
.m-6{margin:1.024rem;}
.text-base{font-size:0.6827rem;line-height:1.024rem;}
.w-100,
.w-100rpx{width:2.1334rem;}
.w-100px{width:4.2667rem;}"
`;

exports[`rpx-tranform > taroH5_750 1`] = `
"/* layer: preflights */
*,::before,::after{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgba(0,0,0,0);--un-ring-shadow:0 0 rgba(0,0,0,0);--un-shadow-inset:var(--un-empty,/*!*/ /*!*/);--un-shadow:0 0 rgba(0,0,0,0);--un-ring-inset:var(--un-empty,/*!*/ /*!*/);--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgba(147,197,253,0.5);}::backdrop{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgba(0,0,0,0);--un-ring-shadow:0 0 rgba(0,0,0,0);--un-shadow-inset:var(--un-empty,/*!*/ /*!*/);--un-shadow:0 0 rgba(0,0,0,0);--un-ring-inset:var(--un-empty,/*!*/ /*!*/);--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgba(147,197,253,0.5);}
/* layer: default */
.m-6{margin:1.024rem;}
.text-base{font-size:0.6827rem;line-height:1.024rem;}
.w-100,
.w-100px,
.w-100rpx{width:2.1334rem;}"
`;

exports[`rpx-tranform > taroWe_375 1`] = `
"/* layer: preflights */
page,::before,::after{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgba(0,0,0,0);--un-ring-shadow:0 0 rgba(0,0,0,0);--un-shadow-inset:var(--un-empty,/*!*/ /*!*/);--un-shadow:0 0 rgba(0,0,0,0);--un-ring-inset:var(--un-empty,/*!*/ /*!*/);--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgba(147,197,253,0.5);}::backdrop{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgba(0,0,0,0);--un-ring-shadow:0 0 rgba(0,0,0,0);--un-shadow-inset:var(--un-empty,/*!*/ /*!*/);--un-shadow:0 0 rgba(0,0,0,0);--un-ring-inset:var(--un-empty,/*!*/ /*!*/);--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgba(147,197,253,0.5);}
/* layer: default */
.m-6{margin:48rpx;}
.text-base{font-size:32rpx;line-height:48rpx;}
.w-100,
.w-100rpx{width:100rpx;}
.w-100px{width:200rpx;}"
`;

exports[`rpx-tranform > taroWe_750 1`] = `
"/* layer: preflights */
page,::before,::after{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgba(0,0,0,0);--un-ring-shadow:0 0 rgba(0,0,0,0);--un-shadow-inset:var(--un-empty,/*!*/ /*!*/);--un-shadow:0 0 rgba(0,0,0,0);--un-ring-inset:var(--un-empty,/*!*/ /*!*/);--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgba(147,197,253,0.5);}::backdrop{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgba(0,0,0,0);--un-ring-shadow:0 0 rgba(0,0,0,0);--un-shadow-inset:var(--un-empty,/*!*/ /*!*/);--un-shadow:0 0 rgba(0,0,0,0);--un-ring-inset:var(--un-empty,/*!*/ /*!*/);--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgba(147,197,253,0.5);}
/* layer: default */
.m-6{margin:48rpx;}
.text-base{font-size:32rpx;line-height:48rpx;}
.w-100,
.w-100px,
.w-100rpx{width:100rpx;}"
`;

exports[`rpx-tranform > uniapp_750 1`] = `
"/* layer: preflights */
uni-page-body,::before,::after{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgba(0,0,0,0);--un-ring-shadow:0 0 rgba(0,0,0,0);--un-shadow-inset:var(--un-empty,/*!*/ /*!*/);--un-shadow:0 0 rgba(0,0,0,0);--un-ring-inset:var(--un-empty,/*!*/ /*!*/);--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgba(147,197,253,0.5);}::backdrop{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgba(0,0,0,0);--un-ring-shadow:0 0 rgba(0,0,0,0);--un-shadow-inset:var(--un-empty,/*!*/ /*!*/);--un-shadow:0 0 rgba(0,0,0,0);--un-ring-inset:var(--un-empty,/*!*/ /*!*/);--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgba(147,197,253,0.5);}
/* layer: default */
.m-6{margin:%?48?%;}
.text-base{font-size:%?32?%;line-height:%?48?%;}
.w-100,
.w-100rpx{width:%?100?%;}
.w-100px{width:100px;}"
`;
